### PR TITLE
Adding ThermoPoromechanics solver

### PR DESCRIPTION
**Pull request by sytuannguyen**
This PR adds a solver for the Thermo-Poromechanics coupled problems. It bases on the existing Poromechanics solver and adds the thermal field to it.

### CODE REVIEW

**Comment by sytuannguyen** [comment]
This PR is closed and will be replaced by another one with updated codebase

**Comment by sytuannguyen** [comment]
I reopen this PR after resolving all the conflicts and updating the THM solver to follow up with the changes in the codebase during the last ten months.

File: host-configs/xraiTotal.cmake [hunk 1]
```diff
@@ -0,0 +1,35 @@
+site_name(HOST_NAME)
```
Comment (herve-gross) [inline host-configs/xraiTotal.cmake:1]: We should leave this file out of the pull request

File: src/coreComponents/linearAlgebra/interfaces/hypre/HypreMGR.cpp [hunk 2]
```diff
@@ -19,7 +19,17 @@
 #include "HypreMGR.hpp"
 
 #include "linearAlgebra/common/common.hpp"
-
+/**
```
Comment (herve-gross) [inline src/coreComponents/linearAlgebra/interfaces/hypre/HypreMGR.cpp:22]: these merge conflict information lines should be removed (many other)

File: src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanicsKernel.hpp [hunk 3]
```diff
@@ -13,17 +13,24 @@
...
 
 /**
- * @file SinglePhasePoroelasticKernel.hpp
+ * @file SinglePhasePoromechanicsKernel.hpp
  */
 
 #ifndef GEOSX_PHYSICSSOLVERS_MULTIPHYSICS_SINGLEPHASEPOROMECHANICSKERNEL_HPP_
 #define GEOSX_PHYSICSSOLVERS_MULTIPHYSICS_SINGLEPHASEPOROMECHANICSKERNEL_HPP_
 
+/**
```
Comment (herve-gross) [inline src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanicsKernel.hpp:22]: merge info - please remove

File: src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanicsKernel.hpp [hunk 4]
```diff
@@ -213,7 +227,12 @@ class SinglePhase :
     }
 
     stack.localFlowDofIndex[0] = m_flowDofNumber[k];
+/**
```
Comment (herve-gross) [inline src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanicsKernel.hpp:230]: merge info - please remove. Also, did you run uncrustify?

File: src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanicsKernel.hpp [hunk 5]
```diff
@@ -290,7 +309,28 @@ class SinglePhase :
 
     // Compute strain increment
     FE_TYPE::symmetricGradient( dNdX, stack.uhat_local, strainIncrement );
-
+/**
```
Comment (herve-gross) [inline src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanicsKernel.hpp:312]: merge info - please remove

**Review by herve-gross — CHANGES_REQUESTED** [review]
There are still a lot of merge conflicts in these files. @francoishamon and @TotoGaz - can you  take a look and help with this PR?

**Review by francoishamon — COMMENTED** [review]
Hello @sytuannguyen it is great to have you back.

After you left, we implemented a thermal option in the flow solvers (single-phase and compositional) with temperature defined at the cell centers. The work in your PR is great, but it is taking an approach that we tried to avoid in the thermal flow solvers, i.e., duplicating the entire solver, and all the assembly kernels, just to add a thermal option.

To make sure that your work fits into the existing thermal framework, we should probably take some time to think about a good design before going forward with your PR, and maybe start with a simple thermal constitutive model? We could discuss that in a meeting if you want, or using the thermoporomechanics slack channel, whatever is easier for you.

**Comment by joshua-white** [comment]
Hi @sytuannguyen.  Yes, in the intervening time @frankfeifan and @CusiniM have already made significant progress on a thermoporomechanics solver.  Let's arrange a meeting to discuss how you can best contribute to the code moving forward.

**Comment by frankfeifan** [comment]
> Hi @sytuannguyen. Yes, in the intervening time @frankfeifan and @CusiniM have already made significant progress on a thermoporomechanics solver. Let's arrange a meeting to discuss how you can best contribute to the code moving forward.

Yes. More specifically, @CusiniM and I have recently added a ThermalSinglePhaseFlow solver by adding the thermal option in the original SinglePhaseFlow solver. I am now also woking on a ThermalPoromechanics solver by sequentially coupling the existing ThermalSinglePhaseFlow solver and the SolidMechanics solver, and about to test it. I agree that we should arrange a meeting to discuss how to proceed.

**Comment by sytuannguyen** [comment]
Hello @herve-gross, @francoishamon, @joshua-white, @frankfeifan thank you very much for your comments! Yes please have rapidely a metting to provide me a good entry point to this subject.

**Pull request closed without merging**